### PR TITLE
Update bubble_sort.md

### PR DESCRIPTION
**Pull request by L-Super**
C++使用`std::swap()`交换数组，同时添加C语言代码，作为原始C++代码的补充

If this PR is related to coding or code translation, please fill out the checklist.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: docs/chapter_sorting/bubble_sort.md [hunk 1]
```diff
@@ -352,7 +365,27 @@ comments: true
...
+            {
+                if (nums[j] > nums[j + 1])
+                {
+                    int temp = nums[j];
+                    nums[j] = nums[j + 1];
+                    nums[j + 1] = temp;
+                    flag = true;
+                }
+            }
+            if(!falg) break;
```
Comment (krahets) [inline docs/chapter_sorting/bubble_sort.md:386]: flag

**Review by krahets — COMMENTED** [review]
Thanks! It is required to add C source files to the `codes/c/...` as well.

Please test the code before creating a PR.

File: docs/chapter_sorting/bubble_sort.md (hunk 1 above)
Comment (L-Super) [reply docs/chapter_sorting/bubble_sort.md]: I'm sorry, didn't notice this mistake

File: codes/c/chapter_sorting/bubble_sort.c [hunk 2]
```diff
@@ -0,0 +1,72 @@
+/**
+ * @file bubble_sort.c
```
Comment (krahets) [inline codes/c/chapter_sorting/bubble_sort.c:2]: Please make the format of the file header match C++.

**Review by krahets — COMMENTED** [review]
Please address the comment.

File: codes/c/chapter_sorting/bubble_sort.c (hunk 2 above)
Comment (L-Super) [reply codes/c/chapter_sorting/bubble_sort.c]: OK. Fixed it.

**Review by L-Super — COMMENTED** [review]
fixed the format problem.

**Comment by L-Super** [comment]
> Thank @L-Super for the code! You are the first contributor writing in C.
> 
> I've committed to a new change. Please check it.
> 
> Looking forward to your next PR.
> 

Sure, It's my pleasure.

**Pull request merged**